Simplified/NYPLUserAccount.swift

```diff
@@ -0,0 +1,402 @@
+    }
+    return nil
+  }
+    
+  var authToken: String? {
+    return NYPLKeychain.shared()?.object(forKey: authTokenKey) as? String ?? nil
+  }
+    
+  var adobeToken: String? {
+    return NYPLKeychain.shared()?.object(forKey: adobeTokenKey) as? String ?? nil
+  }
+    
+  var licensor: [String:Any]? {
+    return NYPLKeychain.shared()?.object(forKey: licensorKey) as? [String:Any] ?? nil
+  }
+    
+  // MARK: - SET
+    
+  @objc(setBarcode:PIN:)
+  func setBarcode(barcode: String, PIN: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(barcode, forKey: barcodeKey)
+    sharedKeychain.setObject(PIN, forKey: PINKey)
+    
+    // make sure to set the barcode related to the current account (aka library)
+    // not the one we just signed in to, because we could have signed in into
+    // library A, but still browsing the catalog of library B.
+    NYPLErrorLogger.setUserID(NYPLUserAccount.sharedAccount().barcode)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+    
+  @objc(setAdobeToken:patron:)
+  func setAdobeToken(token: String, patron: [String : Any]) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(token, forKey: adobeTokenKey)
+    sharedKeychain.setObject(patron, forKey: patronKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setAdobeVender:)
+  func setAdobeVender(vendor: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(vendor, forKey: adobeVendorKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setAdobeToken:)
+  func setAdobeToken(token: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(token, forKey: adobeTokenKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setLicensor:)
+  func setLicensor(licensor: [String : Any]) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
```

Comment:
not sure you need to lock here... all this method is actually doing is issuing a call to NYPLKeychain. Ideally that class should handle its own thread-safety.

Same thing for `setAuthorizationIdentifier` below.